### ISSUE TITLE
Rendering on Mousedown

### ISSUE THREAD

**Issue by saivan** [daniel3735928559/guppy#119]
So I noticed in my profiler, every time I click; it calls a guppy click event handler - which then always re-renders and repaints the page. This is probably not ideal behaviour right? The offending function is `Guppy.mouse_down`, and it seems to go through each guppy instance and re-render it. 

I'm not sure that this is ideal behaviour. I think you are perhaps checking the bounding boxes of the element, which in turn; triggers a repaint of the whole page. Thats probably not ideal behaviour.

## Solution 

Maybe a better way to handle this would be to bind a click handler to the guppy element instead; and only re-render the particular element we clicked on. This would allow you to still have your fancy cursor behaviour, without killing the performance of a page that includes guppy elements.

**Comment by daniel3735928559** [comment]
It looks like this is because of the code to deactivate all the other instances that weren't clicked on, since Guppy.deactivate unconditionally re-renders (I guess "in case it got newly deactivated" or something). This should now be fixed.

**Comment by saivan** [comment]
But this will still run a click handler on any click on the page won't it?

**Comment by daniel3735928559** [comment]
Yeah--this is intentional, sort of like how for a normal textbox, any other click anywhere else on the page causes it to lose focus. Since we don't have the benefit of the browser defocusing us, we check any click and if it missed all guppy instances, it defocuses the one that's currently focused.

**Comment by saivan** [comment]
Yep. But perhaps we should bind the click handler to the element to focus. Then inside the focus handler, we bind a defocus handler to the whole window. That way, you don't run 100 checks every time we click, if we have 100 guppy boxes
